src/test/java/com/puppycrawl/tools/checkstyle/BaseCheckTestSupport.java

```diff
@@ -165,17 +169,45 @@ protected static void verifyAst(String expectedTextPrintFileName, String actualJ
      * @throws Exception if exception occurs during verification.
      */
     protected static void verifyAst(String expectedTextPrintFileName, String actualJavaFileName,
-            boolean withComments) throws Exception {
-        final String expectedContents = new String(Files.readAllBytes(
-            Paths.get(expectedTextPrintFileName)), StandardCharsets.UTF_8)
-            .replaceAll("\\\\r\\\\n", "\\\\n");
+                                    boolean withComments) throws Exception {
+        final String expectedContents = readFile(expectedTextPrintFileName);
+
         final String actualContents = AstTreeStringPrinter.printFileAst(
-                new File(actualJavaFileName), withComments).replaceAll("\\\\r\\\\n", "\\\\n");
+                new File(actualJavaFileName), withComments).replaceAll(CLRF_REGEX, LF_REGEX);
+
+        assertEquals("Generated AST from Java file should match pre-defined AST", expectedContents,
+                actualContents);
+    }
+
+    /**
+     * Verifies the Javadoc Ast generated for the supplied java file against the supplied text file.
+     * @param expectedTextPrintFileName expected text ast tree representation.
+     * @param actualJavaFileName actual text ast tree representation.
+     * @throws Exception if exception occurs during verification.
+     */
+    protected static void verifyJavadocAst(String expectedTextPrintFileName,
```

Comment:
I used ast for `abstract syntax tree` and it is one right ? So should it still be changed ?